### PR TITLE
Support request and limit for pipeline executor

### PR DESCRIPTION
**Pull request by loganhz**
Proposed changes
======
Support config the cpu/memory request and limit in pipeline setting page

Types of changes
======
- Bugfix (non-breaking change which fixes an issue)

Linked Issues
======
https://github.com/rancher/rancher/issues/17569

 

Further comments
======

### CODE REVIEW

File: lib/pipeline/addon/components/settings-git/component.js [hunk 1]
```diff
@@ -20,9 +23,17 @@ export default Component.extend({
...
+    const cpuLimit = get(this, 'settings').findBy('name', 'executor-cpu-limit');
+    const memoryRequest = get(this, 'settings').findBy('name', 'executor-memory-request');
+    const memoryLimit = get(this, 'settings').findBy('name', 'executor-memory-limit');
 
     set(this, 'scale', quota);
     set(this, 'cacerts', cacerts);
+    set(this, 'cpuRequest', convertToMillis(get(cpuRequest, 'value')));
+    set(this, 'cpuLimit', convertToMillis(get(cpuLimit, 'value')));
+    set(this, 'memoryRequest', parseSi(get(memoryRequest, 'value'), 1024) / 1048576);
+    set(this, 'memoryLimit', parseSi(get(memoryLimit, 'value'), 1024) / 1048576);
```
Comment (westlywright) [inline lib/pipeline/addon/components/settings-git/component.js]: `setProperties`

File: lib/pipeline/addon/components/settings-git/template.hbs [hunk 2]
```diff
@@ -57,6 +57,62 @@
       </div>
     </section>
 
+    <section class="box mt-30">
+      <h3>{{t 'authPage.setting.limit.label'}}</h3>
```
Comment (westlywright) [inline lib/pipeline/addon/components/settings-git/template.hbs]: We're mixing single and double quotes in the same html file. We dont have a template linter yet (coming in the next ember upgrade) but when we do the standard is double qoutes in HBS files.

If you are working on an HBS file I am okay if you do a global (file global) find and replace of single quotes to double quotes.

File: lib/pipeline/addon/components/settings-git/template.hbs [hunk 3]
```diff
@@ -57,6 +57,62 @@
...
+      <hr/>
+      <div class="row">
+        <div class="col span-6">
+          <label class="acc-label">{{t 'formSecurity.memoryReservation.label'}}</label>
+          <div class="input-group">
+            {{input-integer min="4" step="1" value=memoryRequest classNames="form-control" placeholder=(t 'formSecurity.memoryReservation.placeholder')}}
```
Comment (westlywright) [inline lib/pipeline/addon/components/settings-git/template.hbs]: Break if props >= 2, for this and all below

**Comment by loganhz** [comment]
Updated. Thanks! @westlywright

**Pull request merged**